File: tot_spec/src/codegen/fixtures/rs_serde/enum.rs
/// Number
#[derive(Debug, serde::Serialize, serde::Deserialize)]
#[serde(tag = "type", content = "payload")]
pub enum Number {
    /// Variant Int64
    Int64(i64),
    /// Variant Float
    Float(f64),
    RealNumber(RealNumber),
}

/// RealNumber
#[derive(Debug, serde::Serialize, serde::Deserialize)]
pub struct RealNumber {
    pub part_0: std::option::Option<f64>,
    pub part_1: std::option::Option<f64>,
}

File: tot_spec/src/codegen/fixtures/typescript/enum.ts

export type
Number =
        // Variant Int64
    { __type: "Int64", payload: bigint }
    |
    |     // Variant Float
    { __type: "Float", payload: number }
    |
    | { __type: "RealNumber", payload: RealNumber }
;


export interface
RealNumber {
    part_0: number | undefined;
    part_1: number | undefined;
}

